homework/index.js

```diff
@@ -23,24 +23,147 @@
+    );
+    addRepoInfoRow(repoTableBody, 'Description', repository.description);
+    addRepoInfoRow(
+      repoTableBody,
+      'Created At',
+      new Date(repository.created_at).toLocaleString('en-GB'),
+    );
+    addRepoInfoRow(
+      repoTableBody,
+      'Updated At',
+      new Date(repository.updated_at).toLocaleString('en-GB'),
+    );
+    addRepoInfoRow(repoTableBody, 'Forks', repository.forks_count);
+    addRepoInfoRow(repoTableBody, 'Watchers', repository.watchers_count);
+  }
+
+  function fetchAndRender(repository) {
+    const rightSide = document.getElementById('right-side');
+    rightSide.innerHTML = '';
+    fetchJSON(repository.contributors_url, (error, contributors) => {
+      if (error) {
+        renderError(error);
+        return;
+      }
+      const contributorsTitle = createAndAppend('h3', rightSide, {
+        text: 'Contributors',
+        class: 'contributors-title',
+      });
+      if (!contributors.length) {
+        contributorsTitle.textContent = 'No Contributor So Far';
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      const contributorsList = createAndAppend('ul', rightSide, {
+        class: 'contributors-list',
+      });
+      contributors.forEach(contributor => {
+        const listItem = createAndAppend('li', contributorsList, {
+          class: 'contributor',
+        });
+        createAndAppend('img', listItem, {
+          class: 'contributor-avatar',
+          src: contributor.avatar_url,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('span', listItem, {
+          class: 'contribution-count',
+          text: contributor.contributions,
+        });
+        listItem.addEventListener('click', () => {
+          window.open(contributor.html_url, '_blank');
+        });
+      });
     });
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  function renderPage(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('h1', header, {
+      class: 'header-title',
+      text: 'Hack Your Future Repositories',
+    });
+    const wrapper = createAndAppend('main', root, {
+      id: 'wrapper',
+    });
+    const select = createAndAppend('select', header);
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        renderError(error);
+        return;
+      }
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
+      repositories.forEach((repository, index) => {
+        createAndAppend('option', select, {
+          value: index,
+          text: repository.name,
+        });
+      });
+      createAndAppend('main', wrapper, {
```

Comment:
Semantically speaking, you should only have one `main` element in your HTML page. I think `section` would be better here.

Reply:
Changed to section tag. Thanks for your advice